src/Data/AbiParser.purs

```diff
@@ -77,18 +81,24 @@ parseBool = string "bool" >>= \_ -> pure SolidityBool
 parseString :: Parser SolidityType
 parseString = string "string" >>= \_ -> pure SolidityString
 
-numberParser :: Parser Int
-numberParser = do
-  n <- fromCharArray <<< fromFoldable <$> many1 anyDigit
-  case fromString n of
-    Nothing -> fail $ "Couldn't parse as Natural : " <> n
+parseDigits :: Parser String
+parseDigits = fromCharArray <<< fromFoldable <$> many1 (try anyDigit)
+
+asInt :: String -> Parser Int
+asInt n = case fromString n of
+    Nothing -> do
+      fail $ "Couldn't parse as Int : " <> n
     Just n' -> pure $ n'
 
 parseBytes :: Parser SolidityType
 parseBytes = do
   _ <- string "bytes"
-  mn <- optionMaybe numberParser
-  pure $ maybe SolidityBytesD SolidityBytesN  mn
+  mns <- optionMaybe parseDigits
```

Comment:
`numberParser` was split so we can treat  fail in parsing digits and in fail in creating int from digits differently.